ui.frontend/src/view/FormFieldBase.js

```diff
@@ -187,15 +187,16 @@ class FormFieldBase extends FormField {
     setFocus(id) {
         const fieldType = this.parentView?.getModel()?.fieldType;
         if (fieldType !== 'form' && this.parentView.setFocus) {
-            this.parentView.setFocus(id);
+            this.parentView.setFocus(this.getId());
         }
-        // If multiple widgets like radio-button or checkbox-group, then focus on the first widget
-        if (this.widget.length > 0) {
-            this.widget[0].focus();
+        this.widget = this.getWidget(); // updating to the latest widget in case of datepicker widget with a formatter
+        if (this.widget instanceof NodeList) {
```

Comment:
Can you check for length, rather than instance of node list ? I would want to reduce dependency on DOM API here

Reply:
when the widget is a dropdown the condition `this.widget.length > 0` is true. Which is not desired. 
this is because (select tag) will have options. 
Where as in case of dropdown and radio it returns an array of Node.

Reply:
Can you add this in the code comments ?